src/v/storage/mvlog/entry_stream_utils.cc

```diff
@@ -0,0 +1,47 @@
+// Copyright 2024 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+#include "storage/mvlog/entry_stream_utils.h"
+
+#include "hashing/crc32.h"
+
+namespace storage::experimental::mvlog {
+
+iobuf entry_header_to_iobuf(const entry_header& hdr) {
+    iobuf b;
+    serde::write(b, hdr.header_crc);
+    serde::write(b, hdr.body_size);
+
+    // NOTE: serde serializes enums as int32_t (serde_enum_serialized_t).
```

Comment:
Perhaps outside the scope of this PR, but should `serde_enum_serialized_t` be a template making use of [underlying_type_t](https://en.cppreference.com/w/cpp/types/underlying_type) instead of forcing a preemptive `static_cast<>` of the enum here?

Reply:
Yeah not a bad idea -- it was a bit surprising and somewhat of a footgun that this is the default behavior for enums. Agreed though, I'm going to punt on this since it's more a pebble in the shoe than a real problem atm